### ISSUE TITLE
Map loading indicator

### ISSUE THREAD

**Issue by npeihl** [Esri/dojo-bootstrap-map-js#83]
Hi @tomwayson 

What are your thoughts on adding a map loading indicator? I'm thinking a spinner like [Spinjs.](http://fgnass.github.io/spin.js/) might be helpful. 

I've added it to my [work-in-progress webapp](http://sjcgis.github.io/traffic-collisions-app/) based on Dojo-bootstrap-map-js. Only problem is I can't get Dojo to build with it. I've tried adding a copyOnly resource tag in app.profile.js, but then I get missing dependencies.

**Comment by tomwayson** [comment]
My thoughts are that it's a good idea - see #57 :)

Spin.js looks pretty cool. 

The issue seems to be that bower is dropping that off directly under the `src` folder and I think it needs to be in sub-folder if you are going to register it as a package.

A quick test would be to put it under `src/spin/spin.js` and then update [these lines](https://github.com/SJCGIS/traffic-collisions-app/blob/89316aa52a4add56377a0409d360314922d10e02/profiles/app.profile.js#L48-51) to

```js
'spin',
```

**Comment by npeihl** [comment]
Arg, I don't know why I didn't already see an issue for that. Sorry.

Bower is actually creating a `spin.js` folder and dropping the library in there. So the structure already looks like this.

```
+---src            
|   \---spin.js
|           .bower.json
|           .gitignore
|           .jshintrc
|           .npmignore
|           bower.json
|           jquery.spin.js
|           LICENSE.txt
|           README.md
|           spin.js
|
```
I wonder if Windows doesn't like the dot in the folder name? Could that be causing trouble?

**Comment by tomwayson** [comment]
Ah, hm that could be it. Did you try renaming it?

**Comment by npeihl** [comment]
Tried renaming it but no good. I checked the build_report and found the following error:

```
error(311) Missing dependency.
	module: spin/jquery.spin; dependency: jquery
	module: spin/jquery.spin; dependency: spin
```

It appears that spin.js can be used as either vanilla javascript or with jquery. I'm using the vanilla flavor and don't have anything requiring jquery. Do you know if it's possible to have the Dojo build ignore the jquery.spin.js file?

**Comment by tomwayson** [comment]
Odd, I don't see anything in your code or [spin.js](https://github.com/fgnass/spin.js/blob/master/spin.js)  that is requiring 'spin/jquery.spin'. Is everything checked into here: https://github.com/SJCGIS/traffic-collisions-app - I'll try to fork and take a look, but no promises, I'm swamped w/ dev summit prep right now.

**Comment by npeihl** [comment]
Hi Tom,

No rush. I appreciate your help and dojo-bootstrap-map-js overall. I'm looking forward to introducing myself at the devsummit next week. 

Thanks. 
Nick 

On March 5, 2015 5:51:40 PM PST, Tom Wayson <notifications@github.com> wrote:
>Odd, I don't see anything in your code or
>[spin.js](https://github.com/fgnass/spin.js/blob/master/spin.js)  that
>is requiring 'spin/jquery.spin'. Is everything checked into here:
>https://github.com/SJCGIS/traffic-collisions-app - I'll try to fork and
>take a look, but no promises, I'm swamped w/ dev summit prep right now.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/Esri/dojo-bootstrap-map-js/issues/83#issuecomment-77492294

-- 
Sent from my Illudium Q36 Explosive Space Modulator

**Comment by tomwayson** [comment]
Nice, you're going to be in Palm Springs? Let's hack on this there! I'm there all week and will be working the Web AppBuilder booth of the ArcGIS Online area in the showcase most of the day Wed/Thu before my late afternoon evening sessions. And of course the Bootstrap Map demo theater is Tues 03:00 PM.

**Comment by npeihl** [comment]
Hi Tom,

Yeah, this will be my second time. Last year at this time I had barely scratched the surface of Dojo and JavaScript. I'd love to get a hack session going with you. Really would like to get started on writing more tests for this, too. 

On March 5, 2015 9:14:47 PM PST, Tom Wayson <notifications@github.com> wrote:
>Nice, you're going to be in Palm Springs? Let's hack on this there! I'm
>there all week and will be working the Web AppBuilder booth of the
>ArcGIS Online area in the showcase most of the day Wed/Thu before my
>late afternoon evening sessions. And of course the Bootstrap Map demo
>theater is Tues 03:00 PM.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/Esri/dojo-bootstrap-map-js/issues/83#issuecomment-77507965

-- 
Sent from my Illudium Q36 Explosive Space Modulator

**Comment by npeihl** [comment]
Hi Tom,

Just an update. Removing the jquery.spin.js file from the spin.js folder makes the build work without error. Still wondering if there is a way for Dojo build to ignore that file or if I need to use a grunt task to delete it prior to building?

**Comment by tomwayson** [comment]
That's great that the build works. So no run time errors? The does spinner works?

I'm still curious about why Dojo wanted to include that file in the build in the first place. You could try to use an `exclude` for that file on the build layer - see the Layer's section of http://dojotoolkit.org/documentation/tutorials/1.10/build/, but if that doesn't work then I'd just use [grunt-contrib-clean](https://github.com/gruntjs/grunt-contrib-clean) for now.

**Comment by tsamaya** [comment]
Hi Nick and Tom,
I commit, on my forked repo, a spinner like Nick did. I changed initialization and the events wiring : I wait the map creation.
The 3.13 upgrade is not working on my laptop on the build task, so I did not Pull Request for the spinner.
Sample is visible at http://gis.tsamaya.net/dojo-bootstrap-map-js/

**Comment by tomwayson** [comment]
Resolved in #86
